### ISSUE TITLE
Cannot obtain per-thread debug files with ConnLogOnly()

### ISSUE THREAD

**Issue by laimaretto** [ktbyers/netmiko#3225]
Hi! This is my first try with `ConnLogOnly()` and netmiko 4.2.0. I'm wiling to store debug info per each individual session, because the classical way of using `logging` mixes all the information in one file, and hence it gets a little problematic to analyze per router info when you have many of them running in parallel.

But even when using `ConnLogOnly()` all of the sessions are also mixed in the same file.

I'm using `from multiprocessing.pool import ThreadPool`  with `apply_async`. Everything works ok, meaning sending commands and receiving output from each individual router on a separate thread, but the debug information with `log_file` is being mixed in only one file. Actually, in the file of the first triggered connection from the thread pool ...

I'm doing as follows....
```
def routerLogin(self, connInfo):
   [...]
   debug = connInfo["ip"] + ".debug"
   conn2rtr = ConnLogOnly(device_type=deviceType, 
                          host=ip, 
                          port=port, 
                          username=tempUser, 
                          password=tempPass, 
                          fast_cli=False, 
                          log_file=debug, 
                          log_level="DEBUG")
```
As you can see, I should have several debug files, but nonetheless, I always get one with the debug info from all the connections ...

However, if I use `ConnectHandler` with `session_log` I do get the separate files per each connection. The problem with this approach is that I cannot control the level of the debugging.
```
def routerLogin(self, connInfo):
   [...]
   debug = connInfo["ip"] + ".debug"
   conn2rtr = ConnectHandler(device_type=deviceType,
                             host=ip, 
                             port=port, 
                             username=tempUser, 
                             password=tempPass, 
                             fast_cli=False, 
                             session_log=debug)`
```
anything to double check? thanks!

**Comment by ktbyers** [comment]
@laimaretto So you are trying to do per-thread logging to separate files. I don't have a solution to this, you would have to research how to do this and see if there is a practical solution.

The `session_log` file is internal to Netmiko and is not using the logging module. The session_log is per Netmiko object so you can have different files per thread.

**Comment by ktbyers** [comment]
Yeah, I totally understand `why` you would want it. I just don't know what the solution is :-)

If you find one, let me know.
